### ISSUE TITLE
1.1.172: False negative for calling positional-only argument by name 

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#3974]
```python
def f(x, /, *args, **kw): ...

f(x=1)   # pyright 1.1.271: error: Expected 1 more positional argument (reportGeneralTypeIssues)
         # pyright 1.1.272: no errors
```
<details><summary>pyright 1.1.271</summary>
<p>

```
Loading configuration file at C:\Users\rsokl\hydra-zen\pyrightconfig.json
Assuming Python version 3.9
Assuming Python platform Windows
stubPath C:\Users\rsokl\hydra-zen\typings is not a valid directory.
Searching for source files
Found 1 source file
pyright 1.1.271
C:\Users\rsokl\hydra-zen\scratch\scratch.py
  C:\Users\rsokl\hydra-zen\scratch\scratch.py:3:5 - error: Expected 1 more positional argument (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 0.799sec
```

</p>
</details>

<details><summary>pyright 1.1.272</summary>
<p>

```console
$ pyright scratch.py
Loading configuration file at C:\Users\rsokl\hydra-zen\pyrightconfig.json
Assuming Python version 3.9
Assuming Python platform Windows
stubPath C:\Users\rsokl\hydra-zen\typings is not a valid directory.
Searching for source files
Found 1 source file
pyright 1.1.272
0 errors, 0 warnings, 0 informations 
Completed in 0.796sec
```

</p>
</details>

I also noticed that the following:

```python
def f(x, /, *args): ...

f(x=1)
```
now raises two, contradictory errors:
```
\scratch.py:3:3 - error: No parameter named "x" (reportGeneralTypeIssues)
\scratch.py:3:1 - error: Argument missing for parameter "x" (reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
@rsokl, out of curiosity, how did you detect a false negative so quickly after the 1.1.272 release? False negatives generally go undetected a long time because they don't show up as errors, so it takes a keen eye to spot them.

**Comment by rsokl** [comment]
> @rsokl, out of curiosity, how did you detect a false negative so quickly after the 1.1.272 release? False negatives generally go undetected a long time because they don't show up as errors, so it takes a keen eye to spot them.

I maintain a collection of pyright-based tests for the type annotations of hydra-zen. Most of my users don't know a thing about types, so I put a lot of effort into ensuring that I am not misleading them with false positives or negatives[^1]. 

The latest release of pyright happened to come out ~20 minutes or so before my nightly tests pulled it down and ran it [(this particular test caught it)](https://github.com/mit-ll-responsible-ai/hydra-zen/blob/main/tests/annotations/declarations.py#L847-L855). So there was some fortunate timing involved here too :)



[^1]: Also, hydra-zen was my first "serious" foray into Python's typing features, so these tests are the only way I could convince myself that I was doing things mostly correctly.

**Comment by erictraut** [comment]
This issue is addressed in pyright 1.1.273, which I just published. It will also be included in a future release of pylance.
